src/ray/gcs/gcs_server/gcs_resource_scheduler.h

```diff
@@ -0,0 +1,186 @@
+// Copyright 2017 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include "absl/container/flat_hash_set.h"
+#include "ray/common/task/scheduling_resources.h"
+#include "ray/gcs/gcs_server/gcs_resource_manager.h"
+
+namespace ray {
+namespace gcs {
+
+// Type of resource scheduling strategy.
+enum SchedulingType {
+  // Places resources across distinct nodes as even as possible.
+  SPREAD = 0,
+  // Places resources across distinct nodes.
+  // It is not allowed to deploy more than one resource on a node.
+  STRICT_SPREAD = 1,
+  // Packs resources into as few nodes as possible.
+  PACK = 2,
+  // Packs resources within one node. It is not allowed to span multiple nodes.
+  STRICT_PACK = 3,
+  SchedulingType_MAX = 4,
```

Comment:
What's the role of `SchedulingType_MAX ` ?

Reply:
This is the maximum value of the enumeration.